utils/find-wasm.js

```diff
@@ -0,0 +1,35 @@
+const util = require('util');
+const glob = util.promisify(require('glob'));
+const fs = require('fs');
+
+
+
+async function findWasm(target) {
+    let res = await glob(`**/${target}/*.wasm`, { 
+        ignore: '**/node_modules/**'
+    });
+    return res;
+}
+
+async function findWasmFile(wasmFile, target) {
+    if (!fs.existsSync(wasmFile)) {
+        if (wasmFile === './out/main.wasm') {
+            let res = await findWasm(target);
```

Comment:
If `./out/main.wasm` is passed, we _ignore_ it and find a separate file instead? 

I don't think that's expectable behavior. I think we need to change/remove the default, rather than use it as a second-tier, undocumented flag.

To me, expectable behavior would be:

* if I provide an option, use it!
* if I don't:
  * use `./out/main.wasm` if it exists, whether for `dev-deploy` or `deploy`. This matches the current experience.
  * look for `contract/target/wasm32-unknown-unknown/${target}/*.wasm` and `contract/build/${target}/*.wasm`, if only one found, use it
    * if multiple wasm files found, throw error message listing these matched wasm files (we can update this later to use an interactive mode to select which you want to deploy)

Reply:
This does just that.  If the file exists, `wasmFile` is returned.  Otherwise if it was `./out/main.wasm` that doesn't exist we check the the other paths using target.

I'll rewrite this in a positive condition to make this clearer.